jupyterlab_gcedetails/src/components/hardware_scaling_form.tsx

```diff
@@ -0,0 +1,243 @@
+  memoryValue: number;
+  attachGpu: boolean;
+  gpuType?: string;
+  gpuCount?: string;
+}
+
+export const STYLES = stylesheet({
+  checkbox: {
+    marginRight: '10px',
+  },
+  checkboxContainer: {
+    paddingTop: '8px',
+    paddingBottom: '10px',
+  },
+  title: {
+    ...BASE_FONT,
+    fontWeight: 500,
+    fontSize: '15px',
+    marginBottom: '5px',
+    ...csstips.horizontal,
+    ...csstips.flex,
+  },
+  formContainer: {
+    padding: '26px 16px 0px 16px',
+  },
+  container: {
+    width: '500px',
+  },
+  description: {
+    paddingBottom: '20px',
+  },
+  topPadding: {
+    paddingTop: '10px',
+  },
+  bottomPadding: {
+    paddingBottom: '10px',
+  },
+});
+
+const NO_ACCELERATOR = ACCELERATOR_TYPES[0].value as string;
+
+export class HardwareScalingForm extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      baseMachineType: machineTypes[0].base.value as string,
+      cpuValue: 0,
+      memoryValue: 0,
+      attachGpu: false,
+      gpuType: NO_ACCELERATOR,
+      gpuCount: '',
+    };
+  }
+
+  private getMachineType = (baseMachineType: string) => {
+    return machineTypes.find(elem => elem.base.value === baseMachineType);
+  };
+
+  render() {
+    const { onDialogClose } = this.props;
+    const { gpuType, gpuCount, attachGpu, baseMachineType } = this.state;
+
+    const onBaseMachineTypeChange = (
+      event: React.ChangeEvent<HTMLSelectElement>
+    ) => {
+      this.setState({
+        baseMachineType: event.target.value,
+        cpuValue: this.getMachineType(event.target.value).configurations[0].cpu,
+        memoryValue: this.getMachineType(event.target.value).configurations[0]
+          .memory,
+      });
+    };
+
+    const onCpuChange = (newValue: number) => {
+      this.setState({
+        cpuValue: newValue,
+      });
+    };
+
+    const onMemoryChange = (newValue: number) => {
+      this.setState({
+        memoryValue: newValue,
+      });
+    };
+
+    const onGpuTypeChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+      this.setState({
+        gpuType: event.target.value,
+        gpuCount: event.target.value
+          ? (ACCELERATOR_COUNTS_1_2_4_8[0].value as string)
+          : '',
+      });
+    };
+
+    const onGpuCountChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+      this.setState({
+        gpuCount: event.target.value,
+      });
+    };
+
+    const onAttachGpuChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+      this.setState({
+        attachGpu: event.target.checked,
+        gpuType: event.target.checked
+          ? (ACCELERATOR_TYPES[1].value as string)
+          : NO_ACCELERATOR,
+        gpuCount: event.target.checked
+          ? (ACCELERATOR_COUNTS_1_2_4_8[0].value as string)
+          : '',
+      });
+    };
+
+    // TODO: Implement submit functionality
+    //eslint-disable-next-line @typescript-eslint/no-empty-function
+    const submitForm = () => {};
```

Comment:
Likewise, I'd suggest lifting this out to a private method and then having the `onClick` property for your submit button call this method.